flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNITCase.java

```diff
@@ -88,13 +94,22 @@ public void testPerJobModeWithDistributedCache() throws Exception {
 			YarnTestCacheJob.getDistributedCacheJobGraph(tmp.newFolder())));
 	}
 
+	@Test
+	public void testPerJobWithProvidedLibDirs() throws Exception {
+		final File tmpFlinkReleaseBinary = temporaryFolder.newFolder("flink-provided-lib");
+		// Prepare a provided flink lib, it should be publicly accessible. So copy it to the /tmp directory.
+		FileUtils.copyDirectory(flinkLibFolder, tmpFlinkReleaseBinary);
+		final Configuration flinkConfig = createDefaultConfiguration(YarnConfigOptions.UserJarInclusion.DISABLED);
+		flinkConfig.set(
```

Comment:
This test passes if we comment out this line,  which is the logic that we want to test.

Reply:
If we comments this line, the `checkStagingDirectory` will go to another if-else branch. So it should pass.
The purpose of this test is to checking whether the staging lib directory exists. If the provided lib dirs is configured, it should not exist. Otherwise, it should exist.

Reply:
Yes, you are right :)